### ISSUE TITLE
web client metrics not working

### ISSUE THREAD

**Issue by JulianBarr** [SeldonIO/seldon-core#1685]
Hi,

I've got this strange problem while using seldon. The web server metrics get exposed correctly but not the web client metrics. I am using seldon engine 1.0.0 but have also tried to update to 1.0.2 and downgrade to 0.5.1 without luck. I am using default setting. Behind the engine is a pipeline of 6 models/transformers. All of them are RESTful. I did not define any custom metrics. The pipeline is working fine, Jaeger shows the trace is successful.  Working in a bank, so it's not convenient to show the config or screenshot. But if you need any more detailed information, please let me know. I once suspected it's due to certain spring boot version implementation. 

By the way, while I was reading the documentation it's seldon.api.executor.cient.requests but in the actual applications.properties, it's seldon.api.engine.client.requests, is this changed?

**Comment by cliveseldon** [comment]
The metrics names will change from 1.1 release if you use the new Golang executor which replaces the Java engine as the service orchestrator.
As for the docs - you need to ensure you choose the version, e.g. https://docs.seldon.io/projects/seldon-core/en/v1.0.2/ for 1.0.2 - there is a drop down at the bottom of the docs page to swicth versions.

You you explain a bit more on what you mean by web server vs web client metrics?

**Comment by JulianBarr** [comment]
Hi Clive,

Thanks for the prompt response.

I see. I am looking forward to the new release. However, I may have to stick to the "engine" release first, since "latest" is not recommended for prime use yet.

I am referring to the documentation at this url:
https://docs.seldon.io/projects/seldon-core/en/v1.0.2/analytics/analytics.html

Please refer to the attached image below.  I am seeing `seldon_api_engine_server_requests_duration_seconds_(bucket,count,sum)`, but not `seldon_api_engine_client_requests_duration_seconds_(bucket,count,sum)`. The latter is not exposed in /prometheus. I am really puzzled about this.

<img width="910" alt="screenshot_seldon_core_analytics" src="https://user-images.githubusercontent.com/63570913/79104361-02e52600-7da1-11ea-9a5b-cbd73cdf6d0d.png">

**Comment by cliveseldon** [comment]
That is strange they should be exposed by micrometer https://github.com/SeldonIO/seldon-core/blob/b4b3cb5f6db42ef5049ef87b1f482adec3af4019/engine/src/main/resources/application.properties#L5

**Comment by JulianBarr** [comment]
I am trying the executor 1.0.2 version to get around with this, but I am stuck at another point.
`time="" level=fatal msg="seldon deployment name must be provided" source="main.go:190"`

I've already enabled executor by setting `seldon.io/executor: true`

Am I missing anything to enable executor instead of engine?

**Comment by cliveseldon** [comment]
How did you install Seldon and which version? The executor is only available in master and will be in 1.1 release (planned this week).
You need to switch on via helm setting. `--set executor.enabled=true`

**Comment by JulianBarr** [comment]
Ok. So If I am using 1.0.2 release, I won't be able to use the executor, even though there's a 1.0.2 docker image for executor. I think I might have been confused by the versioning of 1.0.2. 

So If I want to test, I EITHER checkout the master and build everything from there OR wait for the 1.1 release expected this week. Am I right to say that? In that case, I would wait for the release first.

Sorry for the rude capital case, it only meant to stress.

**Comment by cliveseldon** [comment]
yes, correct. :-)

**Comment by cliveseldon** [comment]
Please try with 1.1.0 release and reopen if an issue.

**Comment by JulianBarr** [comment]
Hi,

I've just tried 1.1.0 release. Yes, it's working. The GO version doesn't have this problem.

**Comment by cliveseldon** [comment]
@JulianBarr great.

**Comment by JulianBarr** [comment]
Thanks @cliveseldon ,

However, I've got another issue with prometheus. Maybe I should open a separate issue instead.

Anyway, allow me describe the problem here first. Prometheus does service discovery with k8s annotations. However, it will check for annotations `prometheus.io/scrape` etc in `service`, I noticed that when creating `Seldondeployment`, seldon only add such annotations to `deployment` but not `service`. This caused prometheus to ignore the endpoint. I once did a test to manually create the `deployment` and `service` with those annotations configured for `service` and it was working. Am I missing anything?
